File: untitled/src/main/scala/spark/Factory/HBaseFactory.scala
package spark.Factory

import org.apache.hadoop.hbase.HBaseConfiguration
import org.apache.hadoop.hbase.mapreduce.TableInputFormat

object HBaseFactory {
  private val master = "hbase.master"
  private val host = "localhost:16030"

  def getConfiguration(inputTable: String) = {
    val config = HBaseConfiguration.create()
    config.set(this.master, this.host)
    config.set(TableInputFormat.INPUT_TABLE, inputTable)
    config.set(TableInputFormat.SCAN, "")
    config
  }
}

File: play-samples-play-scala-hello-world-tutorial/app/factories/HBaseFactory.java
package factories;
import org.apache.hadoop.conf.Configuration;
import org.apache.hadoop.hbase.HBaseConfiguration;
import org.apache.hadoop.hbase.client.Connection;
import org.apache.hadoop.hbase.client.ConnectionFactory;

import java.io.IOException;

public class HBaseFactory {
    public static Connection getConnection() {
        try {
            Configuration config = HBaseConfiguration.create();
            return ConnectionFactory.createConnection(config);
        } catch (IOException e){
            System.out.println(e.toString());
        }
        return null;
    }
}
